### PR TITLE
Spring 07

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

**Review by Baldinini — APPROVED** [review]

File: src/main/java/ua/com/cinema/controllers/UserController.java [hunk 1]
```diff
@@ -21,7 +22,7 @@ public UserController(UserService userService, UserMapper userMapper) {
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto getByEmail(@RequestParam String email) {
+    public UserResponseDto getByEmail(@RequestParam @Email String email) {
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/controllers/UserController.java]: does it work without the steps listed in this article?
https://www.baeldung.com/spring-validate-requestparam-pathvariable

File: src/main/java/ua/com/cinema/model/dto/MovieSessionRequestDto.java [hunk 2]
```diff
@@ -1,8 +1,17 @@
 package ua.com.cinema.model.dto;
 
+import javax.validation.constraints.Min;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/model/dto/MovieSessionRequestDto.java]: wrappers are not supported by this annotation

File: src/main/java/ua/com/cinema/validation/EmailValidator.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_PATTERN = "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@((\\[[0-9]{1,3}"
+            + "\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        if (email == null) {
+            return false;
+        }
+        Pattern pattern = java.util.regex.Pattern.compile(EMAIL_PATTERN);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/validation/EmailValidator.java]:
```suggestion
        Pattern pattern = Pattern.compile(EMAIL_PATTERN);
```

File: src/main/java/ua/com/cinema/validation/Password.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+
+@Constraint(validatedBy = PasswordValidator.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Password {
+
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/validation/Password.java]: redundant empty line

File: src/main/java/ua/com/cinema/validation/PasswordValidator.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package ua.com.cinema.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordValidator implements ConstraintValidator<Password, String> {
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/validation/PasswordValidator.java]:
```suggestion
public class PasswordValidator implements ConstraintValidator<Password, UserRegistrationDto> {
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**